include/envoy/substitution/BUILD

```diff
@@ -0,0 +1,19 @@
+licenses(["notice"])  # Apache 2
```

Comment:
How about include/envoy/formatter and source/common/formatter? I think that's a bit more general and still covers this case? Then we might be able to get the header formatter stuff in there also at some point?

Reply:
OK, rename the folder as formatter, the file is still called substitution_formatter.h. Something like:

include/envoy/formatter/substitution_formatter.h
source/common/formatter/substition_formatter.{h,cc}  substitution_format_string.{h, cc}

Reply:
Done